src/deferredtask.cc

```diff
@@ -2,26 +2,28 @@
 
 namespace exawallet {
 
-DeferredTask::DeferredTask(v8::Isolate* isolate) {
-    resolver.Reset(isolate, v8::Promise::Resolver::New(isolate->GetCurrentContext()).ToLocalChecked());
+DeferredTask::DeferredTask() {
+    v8::Local<v8::Promise::Resolver> value =v8::Promise::Resolver::New(Nan::GetCurrentContext()).ToLocalChecked();
+    resolver.Reset(value);
 }
 
 DeferredTask::~DeferredTask() {
     resolver.Reset();
 }
 
-void DeferredTask::Resolve(v8::Isolate* isolate, v8::Local<v8::Value> value) {
-    resolver.Get(isolate)->Resolve(isolate->GetCurrentContext(), value);
+void DeferredTask::Resolve(v8::Local<v8::Value> value) {
+    Nan::New(resolver)->Resolve(Nan::GetCurrentContext(), value);
 }
 
-void DeferredTask::Reject(v8::Isolate* isolate) {
-    resolver.Get(isolate)->Reject(isolate->GetCurrentContext(), v8::String::NewFromUtf8(isolate, errorString.c_str()));
+void DeferredTask::Reject() {
+    v8::Local<v8::Context> context = Nan::GetCurrentContext();
```

Comment:
why did you rewrite it?

Reply:
There were some compilation errors I managed to fix introducing explicit type for value variable.